### PR TITLE
feat: rewrite route with http logger plugin with ginkgo && fix: base.CleanAPISIXErrorLog api 

### PR DESCRIPTION
**Pull request by bisakhmondal**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [x] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Please update this section with detailed description.

**Related issues**
#1500 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: api/test/e2enew/base/base.go [hunk 1]
```diff
@@ -248,8 +248,9 @@ func CleanAPISIXErrorLog() {
 	pwd := string(pwdByte)
 	pwd = strings.Replace(pwd, "\n", "", 1)
 	pwd = pwd[:strings.Index(pwd, "/e2e")]
-	cmd = exec.Command("sudo", "echo", " > ", pwd+"/docker/apisix_logs/error.log")
-	_, err = cmd.CombinedOutput()
+	cmdStr := "echo | sudo tee " + pwd+"/docker/apisix_logs/error.log"
+	cmd = exec.Command("bash", "-c", cmdStr)
+	_, err = cmd.Output()
```
Comment (bisakhmondal) [inline api/test/e2enew/base/base.go:253]: FYI, earlier the cleanup was not taking place and it got clear during the execution of [this line](https://github.com/apache/apisix-dashboard/pull/1755/files#diff-fda12d792401cbe1075880088e985ad29db054f11c5cec63b0dd68da1298b454R198), as for this substring not to be present the cleanup must be done.
Reference: [stackoverflow](https://stackoverflow.com/a/30329351)

File: api/test/e2enew/route/route_with_plugin_http_logger_test.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+		// verify http logger by checking log
+		logContent := base.ReadAPISIXErrorLog()
+		gomega.Expect(logContent).ShouldNot(gomega.ContainSubstring("Batch Processor[http logger] successfully processed the entries"))
+
+		// clean log
+		base.CleanAPISIXErrorLog()
+	})
+
+	table.DescribeTable("test route with log plugin",
```
Comment (starsz) [inline api/test/e2enew/route/route_with_plugin_http_logger_test.go:204]: This description confused me.
Maybe delete the route?

Comment (nic-chen) [reply api/test/e2enew/route/route_with_plugin_http_logger_test.go:204]: we could use `clean test data`

**Review by nic-chen — APPROVED** [review]
LGTM expect some descriptions.

File: api/test/e2enew/route/route_with_plugin_http_logger_test.go (hunk 2 above)
Comment (bisakhmondal) [reply api/test/e2enew/route/route_with_plugin_http_logger_test.go:204]: Awesome!!

**Review by Jaycean — APPROVED** [review]
LGTM

**Review by gxthrj — APPROVED** [review]
LGTM

**Review by starsz — APPROVED** [review]

**Pull request merged**